### PR TITLE
Remove get_vertex_local_index

### PR DESCRIPTION
**Pull request by michalhabera**
Small performance improvements for `create_entity_permutations`.

### CODE REVIEW

File: cpp/dolfinx/mesh/Mesh.cpp [hunk 1]
```diff
@@ -448,12 +456,20 @@ void Mesh::create_entity_permutations() const
           // the lowest numbered vertex to the highest numbered vertex
           if (d == 1)
           {
-            auto vertices = facet.entities(0);
-            const int e_vertices[2]
-                = {cell.get_vertex_local_index(vertices[0]),
-                   cell.get_vertex_local_index(vertices[1])};
+            const auto vertices = facet.entities(0);
```
Comment (garth-wells) [inline cpp/dolfinx/mesh/Mesh.cpp]: Use
```
auto vertices = c_f_v.links(f);
```
where `c_f_v` is the face-vertex connectivity.

File: cpp/dolfinx/mesh/Mesh.cpp [hunk 2]
```diff
@@ -462,11 +478,18 @@ void Mesh::create_entity_permutations() const
             // lower number than the next vertex clockwise. Find the index of
             // the lowest numbered vertex
             rots = 0;
-            auto vertices = facet.entities(0);
-            const int e_vertices[3]
-                = {cell.get_vertex_local_index(vertices[0]),
-                   cell.get_vertex_local_index(vertices[1]),
-                   cell.get_vertex_local_index(vertices[2])};
+            const auto vertices = facet.entities(0);
```
Comment (garth-wells) [inline cpp/dolfinx/mesh/Mesh.cpp]: Use
```
auto vertices = c_f_v.links(f);
```
where c_f_v is the face-vertex connectivity.

File: cpp/dolfinx/mesh/Mesh.cpp [hunk 3]
```diff
@@ -486,13 +509,19 @@ void Mesh::create_entity_permutations() const
...
+  }
   // If the cell is a quad, hex or interval
   else
   {
+    // Store local vertex indices here
+    std::array<std::size_t, 4> e_vertices;
+
     for (int cell_n = 0; cell_n < num_cells; ++cell_n)
     {
       const mesh::MeshEntity cell(*this, tdim, cell_n);
```
Comment (garth-wells) [inline cpp/dolfinx/mesh/Mesh.cpp]: Is this needed?

File: cpp/dolfinx/mesh/Mesh.cpp [hunk 4]
```diff
@@ -486,13 +509,19 @@ void Mesh::create_entity_permutations() const
...
+      // Fetch all vertices for the cell here, for performance reasons
+      const auto cell_vertices = this->topology().connectivity(tdim, 0)->links(cell_n);
```
Comment (garth-wells) [inline cpp/dolfinx/mesh/Mesh.cpp]: Add `assert`.

File: cpp/dolfinx/mesh/Mesh.cpp [hunk 5]
```diff
@@ -509,12 +538,18 @@ void Mesh::create_entity_permutations() const
           // the lowest numbered vertex to the highest numbered vertex
           if (d == 1)
           {
-            auto vertices = facet.entities(0);
-            const int e_vertices[2]
-                = {cell.get_vertex_local_index(vertices[0]),
-                   cell.get_vertex_local_index(vertices[1])};
+            const auto vertices = facet.entities(0);
```
Comment (garth-wells) [inline cpp/dolfinx/mesh/Mesh.cpp]: See above on using connectivity directly.

File: cpp/dolfinx/mesh/Mesh.cpp [hunk 6]
```diff
@@ -525,12 +560,17 @@ void Mesh::create_entity_permutations() const
             // number than the next vertex clockwise. Find the index of the
             // lowest numbered vertex
             int num_min = -1;
-            auto vertices = facet.entities(0);
-            const int e_vertices[4]
-                = {cell.get_vertex_local_index(vertices[0]),
-                   cell.get_vertex_local_index(vertices[1]),
-                   cell.get_vertex_local_index(vertices[2]),
-                   cell.get_vertex_local_index(vertices[3])};
+            const auto vertices = facet.entities(0);
```
Comment (garth-wells) [inline cpp/dolfinx/mesh/Mesh.cpp]: Se above.

File: cpp/dolfinx/mesh/Mesh.cpp [hunk 7]
```diff
@@ -429,16 +429,24 @@ void Mesh::create_entity_permutations() const
...
+    // Store local vertex indices here
+    std::array<std::size_t, 3> e_vertices;
+
     for (int cell_n = 0; cell_n < num_cells; ++cell_n)
     {
       const mesh::MeshEntity cell(*this, tdim, cell_n);
+
+      // Fetch all vertices for the cell here, for performance reasons
+      const auto cell_vertices = this->topology().connectivity(tdim, 0)->links(cell_n);
+
```
[the remark this answers is not in the document]
Comment (michalhabera) [reply cpp/dolfinx/mesh/Mesh.cpp]: Assert doesn't know how to cast Eigen Array to bool.

[the remark this answers is not in the document]
Comment (garth-wells) [reply cpp/dolfinx/mesh/Mesh.cpp]: It should be `assert(this->topology().connectivity(tdim, 0);` or just  `assert(_topology->connectivity(tdim, 0);`.

**Pull request merged**